cli/tests/pcluster/utils/test_pcluster_utils.py

```diff
@@ -1008,3 +967,129 @@ def test_get_ebs_snapshot_info(boto3_stubber, snapshot_id, raise_exceptions, err
         with pytest.raises(SystemExit, match=error_message) as sysexit:
             utils.get_ebs_snapshot_info(snapshot_id, raise_exceptions=raise_exceptions)
             assert_that(sysexit.value.code).is_not_equal_to(0)
+
+
+class TestCache:
+    invocations = []
+
+    @pytest.fixture(autouse=True)
+    def clear_invocations(self):
+        self.invocations.clear()
+
+    @pytest.fixture
+    def disabled_cache(self):
+        os.environ["PCLUSTER_CACHE_DISABLED"] = "true"
+        yield
+        del os.environ["PCLUSTER_CACHE_DISABLED"]
+
+    @utils.Cache.cached
+    def _cached_method_1(self, arg1, arg2):
+        TestCache.invocations.append((arg1, arg2))
+        return arg1, arg2
+
+    @utils.Cache.cached
+    def _cached_method_2(self, arg1, arg2):
+        TestCache.invocations.append((arg1, arg2))
+        return arg1, arg2
+
+    def test_cached_method(self):
+        for _ in range(0, 2):
+            assert_that(self._cached_method_1(1, 2)).is_equal_to((1, 2))
+            assert_that(self._cached_method_2(1, 2)).is_equal_to((1, 2))
+            assert_that(self._cached_method_1(2, 1)).is_equal_to((2, 1))
+            assert_that(self._cached_method_1(1, arg2=2)).is_equal_to((1, 2))
+            assert_that(self._cached_method_1(arg1=1, arg2=2)).is_equal_to((1, 2))
+
+        assert_that(self.invocations).is_length(5)
```

Comment:
what about checking the content or the size of the `Cache.caches` ?

Reply:
Two reasons why I prefer not to do that:
1. caches is part of the internal implementation of the caching mechanism not something the class exposes (actually let me mark it as such). Having a unit test depend on the internal implementation of the caching mechanism makes the test invalid even if the class interface does not change.
2. new items are added to the caches array when the annotated function is loaded and not when it is executed for the first time, which means the number to assert depends on the number of imported functions that are annotated with `@Cache.cached`